test/expected/bgw_db_scheduler.out

```diff
@@ -100,11 +100,45 @@ SELECT * FROM sorted_bgw_log;
       0 |         0 | DB Scheduler     | [TESTING] Wait until 50000, started at 0
 (1 row)
 
+--
+-- Test running the scheduler with a job marked as unscheduled
+--
+SELECT ts_bgw_params_reset_time();
+ ts_bgw_params_reset_time 
+--------------------------
+ 
+(1 row)
+
+SELECT insert_job('unscheduled', 'bgw_test_job_1', INTERVAL '100ms', INTERVAL '100s', 5, INTERVAL '1s',scheduled:= false);
+ insert_job 
+------------
+ 
+(1 row)
+
+SELECT ts_bgw_db_scheduler_test_run_and_wait_for_scheduler_finish(50);
+ ts_bgw_db_scheduler_test_run_and_wait_for_scheduler_finish 
+------------------------------------------------------------
+ 
+(1 row)
+
+-- empty
+SELECT * FROM _timescaledb_internal.bgw_job_stat;
+ job_id | last_start | last_finish | next_start | last_successful_finish | last_run_success | total_runs | total_duration | total_successes | total_failures | total_crashes | consecutive_failures | consecutive_crashes 
+--------+------------+-------------+------------+------------------------+------------------+------------+----------------+-----------------+----------------+---------------+----------------------+---------------------
+(0 rows)
+
+SELECT delete_job(1000);
+ delete_job 
+------------
+ 
+(1 row)
+
 --
```

Comment:
also need a test that shows if the state is changed from scheduled = false to scheduled = true, the job gets picked up.
And vice-versa. You probably need some additional code for this to work.

Reply:
this is already handled, the job list is updated on cache invalidations

Reply:
You need an API in job.c to update the scheduled state . IIRC invalidation callbacks are triggered only in that case.

Reply:
well i didnt wanna make api changes in this PR, i was gonna add the scheduled flag to alter_job_schedule in a different PR, do you think this should be part of this PR?

Reply:
I think it makes sense to add all the "scheduled" changes as part of this PR.

Reply:
i disagree i think it makes sense to have all the api changes together especially since there are probably more changes to alter_job_schedule